HIP/hip-583.md

```diff
@@ -2,221 +2,268 @@
+TODO: Image needs to be updated
+![lazy-create-with-address.png](../assets/hip-583/lazy-create-with-address.png)
 
-Since Hedera accounts require keys to ensure security, a matching `ECDSA` `public-key` must be provided in a future signed transaction issued by the owner of the `public-address`. Due to the nature of `ECDSA` keys the `public-key` and `public-address` may be extracted from the signature of a transaction signed by the `ECDSA` private key that owns the new account. This process will complete the creation of the account, taking it from a `Hollow Account` to a normal complete account with a valid `public-key`.
-Note, in this case the final value of the account alias will remain as the `public-address` and not the `public-key` as it would be in the `auto-create` case.
+When a user transfers value (hbars or tokens), they may address their destination account using the standard
+account ID format (`shard.realm.num`), or by using an alias format (`shard.realm.alias`). If the alias is an
+`evm address alias`, then the consensus node will first check to see if such an account exists. If it does,
+then it will use the normal rules for transferring the hbars or tokens (including checking `receiverSigRequired`,
+and all other rules). If the account *does not* exist, then the account will be auto-created with the same rules
+specified in `HIP-32`, except that no key will be present on the newly created account. An account created in this
+manner will be known as a `Hollow Account` and will only be able to receive HBAR and applicable token transfers, but
+cannot take part in transactions submitted by others requiring its signature.
 
-As an illustration, a wallet or exchange could create and fund a new Hedera account based on the `public-address` alone. Once the transaction goes through the owner of the `public-address` user may use a wallet of their choice to submit a signed transaction (e.g. `CryptoTransfer`, `ContractCreate`, `ContractCall`, `EthereumTransaction` or `TokenAssociation`) to claim ownership of the account and carry out the desired transaction. The ownership claim and account completion will be seamless and transparent to the user.
+Since Hedera accounts require keys to ensure security, an `ECDSA` key matching the derived `evm address alias` must be
+provided in a future signed transaction issued by the owner of the account to take control of the account. Due to the
+nature of `ECDSA secp256k1` keys, the public key can be extracted from the signature and the signed bytes. It can then
+go through the derivation function to generate the associated `evm address`, which can be compared with the `alias` to
+prove the match and account ownership. This process will complete the creation of the account, taking it from a
```

Comment:
Just to note an implementation detail for HAPI transactions: for the completion of the hollow account currently we are expecting a full-prefix signature (i.e. we have the entire public key in the signature) instead of deriving it in a way similar to `ecRecover`.

Reply:
OK, I left this language the same as was in the previous version, so it sounds like we did not implement it as specified in the HIP? From @bugbytesinc comment above, it sounds like we have an issue here to resolve.

Reply:
Yeah let's call this nuance out.
It doesn't affect work as this is the case when utilizing SDK's but it's good to note.

Reply:
@rbair23 there are a couple of different scenarios for hollow account completion.
1. In the context of an `EthereumTransaction` flow we do extract the public `ECDSA` key from the raw Ethereum signed transaction and use that key for the completion of the hollow account. Reference: [SigsAndPayerKeyScreen](https://github.com/hashgraph/hedera-services/blob/ac9d19c615dfbbd9d3876cfe1fcb017373fb6ae6/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/logic/SigsAndPayerKeyScreen.java#L124). So that part is correct and implemented.
3. In the context of a `HAPI` transaction flow. There is no `ecrecover`ing for HAPI transactions. Instead, we look at the `payer` of incoming transactions, and try to match the full-prefix public key to the hollow account `EVM address alias`. I think @MiroslavGatsanoga was referencing this flow.

It has to be the payer. It was implemented like this due to performance considerations. Searching through the whole signature map of each incoming transaction and trying to find matching hollow accounts… is this something we want to do?

Reply:
With https://github.com/hashgraph/hedera-services/pull/4990 we're moving the account finalization logic to prehandle and updating it so that _any_ full-prefixed ECDSA public key from the sig map will be matched to _any_ existing hollow account in Hedera and used to finalize it.

Reply:
It is my understanding from the preliminary implementation (on previewnet) that this is ONLY implemented for PAYER accounts.  Meaning that only the payer signature will be checked for ecrecover, any other signature will not be examined in this process.  So if you have a service that has nominated itself to be the payer in a TX which is still _also_ signed_ by an secp256k1 matching a hollow account participating in the TX, the TX _will fail because the secp256k1 key is not the payer_

Can someone please clarify the intent of the HIP and how it will be implemented?